PiwikTracker/PiwikUserDefaults.swift

```diff
@@ -0,0 +1,94 @@
+import Foundation
+
+/// PiwikUserDefaults is a wrapper for the UserDefaults with properties
+/// mapping onto values stored in the UserDefaults.
+/// All getter and setter are sideeffect free and automatically syncronize
+/// after writing.
+internal class PiwikUserDefaults {
+    internal static let standard = PiwikUserDefaults()
+    internal let userDefaults = UserDefaults.standard
+    
+    internal var totalNumberOfVisits: UInt32 {
+        get {
+            return UInt32(self.userDefaults.integer(forKey: PiwikUserDefaults.Constants.UserDefaultTotalNumberOfVisitsKey))
+        }
+        set {
+            self.userDefaults.set(newValue, forKey: PiwikUserDefaults.Constants.UserDefaultTotalNumberOfVisitsKey)
+            self.userDefaults.synchronize()
+        }
+    }
+    
+    internal var firstVisit: Date? {
+        get {
+            return self.date(forKey: PiwikUserDefaults.Constants.UserDefaultFirstVistsTimestampKey)
+        }
+        set {
+            self.userDefaults.set(newValue, forKey: PiwikUserDefaults.Constants.UserDefaultFirstVistsTimestampKey)
+            self.userDefaults.synchronize()
+        }
+    }
+    
+    internal var previousVisit: Date? {
+        get {
+            return self.date(forKey: PiwikUserDefaults.Constants.UserDefaultPreviousVistsTimestampKey)
+        }
+        set {
+            self.userDefaults.set(newValue, forKey: PiwikUserDefaults.Constants.UserDefaultPreviousVistsTimestampKey)
+            self.userDefaults.synchronize()
+        }
+    }
+    
+    internal var currentVisit: Date? {
+        get {
+            return self.date(forKey: PiwikUserDefaults.Constants.UserDefaultCurrentVisitTimestampKey)
+        }
+        set {
+            self.userDefaults.set(newValue, forKey: PiwikUserDefaults.Constants.UserDefaultCurrentVisitTimestampKey)
+            self.userDefaults.synchronize()
+        }
+    }
+    
+    internal var optOut: Bool {
+        get {
+            return self.userDefaults.bool(forKey: PiwikUserDefaults.Constants.UserDefaultOptOutKey)
+        }
+        set {
+            self.userDefaults.set(newValue, forKey: PiwikUserDefaults.Constants.UserDefaultOptOutKey)
+            self.userDefaults.synchronize()
+        }
+    }
+    
+    internal func clientId(prefix: String) -> String? {
```

Comment:
The `prefix` is an old piece of code meant to enable two PiwikTracker for two different backends with two different site ids. So as a prefix the site id could be used.

I removed it for now.